src/coreclr/jit/emitxarch.cpp

```diff
@@ -1815,12 +1815,24 @@ const emitJumpKind emitReverseJumpKinds[] = {
 
 inline bool emitInstHasNoCode(instruction ins)
 {
-    if (ins == INS_align)
+    switch (ins)
     {
-        return true;
-    }
+        case INS_align:
+        {
+            return true;
+        }
 
-    return false;
+        case INS_mov:
```

Comment:
It seems very odd that `emitInstHasNoCode` would return true for `INS_mov`, since it's only some (small number) of mov's that are elided. E.g., `emitter::getInsExecutionCharacteristics()` would have a case for it, instead of a special case in `insEvaluateExecutionCost`.

nit: you don't need all those per-case braces.

Reply:
Fixed by defining a `INS_mov_eliminated` pseudo-op